aptos-move/aptos-gas/src/misc.rs

```diff
@@ -166,27 +168,49 @@ impl<'a> ValueVisitor for AbstractValueSizeVisitor<'a> {
 
     #[inline]
     fn visit_vec_u8(&mut self, _depth: usize, vals: &[u8]) {
-        self.size += self.params.per_u8_packed * NumArgs::new(vals.len() as u64);
+        let mut size = self.params.per_u8_packed * NumArgs::new(vals.len() as u64);
```

Comment:
Those can never overflow because of some known size restriction of `vals.len()`?

Reply:
Both Add and Mul are saturating here so we should be fine